public/app/plugins/panel/table/transformers.ts

```diff
@@ -209,8 +211,11 @@ transformers['json'] = {
       const tableCol: any = { text: column.text };
 
       // if filterable data then set columns to filterable
-      if (data.length > 0 && data[0].filterable) {
-        tableCol.filterable = true;
+      if (data.length > 0) {
+        const dataFrame = toDataFrame(data[0]);
+        if (dataFrame.meta && dataFrame.fields[0].config.filterable) {
+          tableCol.filterable = true;
+        }
```

Comment:
All of this feels really messy. Suggestion of improvement very welcome.

Reply:
kill them all!!!!   I hope we can replace all of this with a new table panel and using 'transformers' rather than the one-off transformers in table panel that are calculated a few ways :(

Reply:
Yeah, but the point of this PR is to mainly resolve the critical bug. We have to create a separate issue to do bigger enhancement/migrations eventually.